### PR TITLE
ARS Enhancements 

### PR DESCRIPTION
**Pull request by JaiOCP**
New global attributes added to
1. Enable/DIsable IPv4 traffic
2. Enable/Disable IPv6 traffic

SAI_ACL_ENTRY_ATTR_ACTION_DISABLE_ARS_FORWARDING attribute modified as a data type bool.
Default value is ARS forwarding is enabled. Setting bool to true means that ARS forwarding is disabled.
1.  ACL ARS forwarding Enable and NHG with ARS binding: ARS forwarding takes effect
2. ACL ARS forwarding Enable and NHG without ARS binding: ACL entry is a noop and NHG forwarding takes effect.
3. ACL ARS forwarding Disable and NHG with ARS binding: NHG forwarding takes effect
4. ACL ARS forwarding Disable and NHG without ARS binding: ACL entry is a noop and NHG forwarding takes effect.

### CODE REVIEW

**Review by kcudnik — APPROVED** [review]

**Review by rck-innovium — APPROVED** [review]

**Comment by rck-innovium** [comment]
Please deprecate SAI\_ACL\_ACTION\_TYPE\_DISABLE\_ARS\_FORWARDING

File: inc/saiacl.h [hunk 1]
```diff
@@ -3204,10 +3208,19 @@ typedef enum _sai_acl_entry_attr_t
...
     SAI_ACL_ENTRY_ATTR_ACTION_CHAIN_REDIRECT = SAI_ACL_ENTRY_ATTR_ACTION_START + 0x38,
 
+    /**
+     * @brief Disable ARS forwarding for a flow
+     *
+     * @type sai_acl_action_data_t bool
+     * @flags CREATE_AND_SET
+     * @default disabled
+     */
+    SAI_ACL_ENTRY_ATTR_ACTION_DISABLE_ARS = SAI_ACL_ENTRY_ATTR_ACTION_START + 0x39,
```
Comment (rlhui) [inline inc/saiacl.h:3218]: add comment that this attribute for ACL is applicable only if the global ARS is enabled.

**Comment by JaiOCP** [comment]
> Please deprecate SAI_ACL_ACTION_TYPE_DISABLE_ARS_FORWARDING

Per community meeting, re-purpose the attribute instead of deprecating it.

**Review by shri-khare — APPROVED** [review]

**Review by kcudnik — APPROVED** [review]

**Comment by JaiOCP** [comment]
Kamil,Can you please merge this PR. Looks like all have approved it.On Mar 7, 2024, at 11:27 PM, Kamil Cudnik ***@***.***> wrote:﻿
@kcudnik approved this pull request.



—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you modified the open/close state.Message ID: ***@***.***>

This electronic communication and the information and any files transmitted with it, or attached to it, are confidential and are intended solely for the use of the individual or entity to whom it is addressed and may contain information that is confidential, legally privileged, protected by privacy laws, or otherwise restricted from disclosure to anyone else. If you are not the intended recipient or the person responsible for delivering the e-mail to the intended recipient, you are hereby notified that any use, copying, distributing, dissemination, forwarding, printing, or copying of this e-mail is strictly prohibited. If you received this e-mail in error, please return the e-mail to the sender, delete it from your computer, and destroy any printed copy of it.

**Pull request merged**